gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/mapper/ApiMapper.java

```diff
@@ -240,4 +237,77 @@ private String toApiDefinition(final String apiId, final NewApiEntity newApiEnti
             throw new TechnicalManagementException("An error occurs while trying to parse API definition " + jse);
         }
     }
+
+    private String toApiDefinition(final UpdateApiEntity updateApiEntity, final ApiEntity existingApiEntity) {
+        try {
+            io.gravitee.definition.model.v4.Api apiDefinition = new io.gravitee.definition.model.v4.Api();
+
+            // Must not update Type
+            apiDefinition.setType(existingApiEntity.getType());
```

Comment:
We should explicitly compare previous and new type, and thrown an exception if the type has been updated instead of ignoring the new value.

Reply:
👍🏻